packages/studio/src/components/EditStreamScopeModal.tsx

```diff
@@ -0,0 +1,98 @@
+import { useCallback, useMemo, useState } from "react";
+import DialogModal from "./common/DialogModal";
+import StreamScopeInput from "./StreamScopeInput";
+import { ResponseType, StreamScope } from "@yext/studio-plugin";
+import useStudioStore from "../store/useStudioStore";
+import { isEqual } from "lodash";
+import { toast } from "react-toastify";
+
+export interface EditStreamScopeModalProps {
+  pageName: string;
+  isOpen: boolean;
+  handleClose: () => void;
+}
+
+export default function EditStreamScopeModal(props: EditStreamScopeModalProps) {
+  const { pageName, isOpen, handleClose } = props;
+  const [streamScope, updateStreamScope, generateTestData] = useStudioStore(
+    (store) => [
+      store.pages.pages[pageName].pagesJS?.streamScope,
+      store.pages.updateStreamScope,
+      store.actions.generateTestData,
+    ]
+  );
+
+  const originalScope = useMemo(() => streamScope ?? {}, [streamScope]);
+  const [selectedScope, setSelectedScope] = useState(originalScope);
+
+  const onConfirm = useCallback(async () => {
+    updateStreamScope(pageName, selectedScope);
+    const regenerateTestData = async () => {
+      const response = await generateTestData();
+      if (response.type === ResponseType.Error) {
+        toast.error(
+          "Error generating test data, but entity page settings were still updated."
+        );
+      }
+    };
+    await regenerateTestData();
+    handleClose();
+  }, [
+    generateTestData,
+    handleClose,
+    pageName,
+    selectedScope,
+    updateStreamScope,
+  ]);
+
+  const updateSelection = useCallback(
+    (streamScopeField: keyof StreamScope) => (selectedIds: string[]) => {
+      if (selectedIds.length) {
+        setSelectedScope({
+          [streamScopeField]: selectedIds,
+        });
+      } else {
+        setSelectedScope({});
+      }
+    },
+    []
+  );
+
+  const hasNoChanges = useMemo(
+    () => isEqual(selectedScope, originalScope),
```

Comment:
having the same filters but in different order is still considered different. is that OK?

Reply:
How much work would it be for that not to be considered different?

Reply:
gave it a little thought, changed it so order doesn't matter 👍